code/data_structures/linked_list/linked_lists_operations/Reverse_recursion_linked_list.cpp

```diff
@@ -0,0 +1,92 @@
+#include<iostream>
+#include<stack>
+// Part of Cosmos by OpenGenus Foundation
+using namespace std;
+
+class node{
+public:
+    int data;
+    node* next;
+
+public:
+    node(int d){
+        data=d;
+        next=NULL;
+    }
+};
+
+void print(node*head){
+    while(head!=NULL){
+        cout<<head->data<<"-->";
+        head=head->next;
+    }
+}
+
+void insertAtHead(node*&head,int d){
+    node* n=new node(d);
+    n->next=head;
+    head=n;
+}
+
+void insertInMiddle(node*&head,int d,int p){
+    if(p==0){
+        insertAtHead(head,d);
+    }
+    else{
+        //assuming p<=length of LL
+        int jump=1;
+        node*temp=head;
+        while(jump<=p-1){
```

Comment:
Use the condition `< p` instead. While yours is equivalent, it is also unconventional. Also, this loop could be converted into a for loop.